app/src/main/java/com/pokescanner/MapsActivity.java

```diff
@@ -944,11 +944,17 @@ public void onLowMemory() {
         super.onLowMemory();
         cleanEntireMap();
     }
-    public boolean moveCameraToCurrentPosition() {
+    public boolean moveCameraToCurrentPosition(boolean zoom) {
         LatLng GPS_LOCATION = getCurrentLocation();
         if (GPS_LOCATION != null) {
             if (mMap != null) {
-                this.mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(GPS_LOCATION, 15));
+                if (zoom) {
+                    this.mMap.animateCamera(CameraUpdateFactory.newLatLng(GPS_LOCATION));
+                    this.mMap.animateCamera(CameraUpdateFactory.zoomTo(15));
```

Comment:
These are separated to separate calls in order to not have the re-centering that happens on initial load to be zoom in interrupted.